### PR TITLE
Save data input form

### PR DESCRIPTION
**Pull request by tinashechiraya**
https://github.com/kartoza/miniSASS/assets/70011086/26ced76b-c9c4-41f6-a5e8-b2e9d4999f76

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 18 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 12 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 37 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 31 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 26 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 30 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 17 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 51 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 28 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 33 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 48 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 7 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 0 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 9 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 19 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 29 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 26 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 44 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 44 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 39 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 21 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 22 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 34 sec

File: django_project/monitor/observation_views.py [hunk 1]
```diff
@@ -11,6 +11,118 @@
...
+            except Sites.DoesNotExist:
+                # If it doesn't exist, create a new site
+                site_name = datainput.get('siteName', '')
+                river_name = datainput.get('riverName', '')
+                description = datainput.get('siteDescription', '')
+                river_cat = datainput.get('rivercategory', '')
+                # longitude = Decimal(datainput.get('longitude',0))
+                # latitude = Decimal(datainput.get('latitude',0))
+                # TODO still to merge with Irwan for saving location
+                longitude = Decimal('30.12345')
```
Comment (meomancer) [inline django_project/monitor/observation_views.py]: Please use datainput.get('longitude', 0)

File: django_project/monitor/observation_views.py [hunk 2]
```diff
@@ -11,6 +11,118 @@
...
+                latitude = Decimal('-25.67890')
```
Comment (meomancer) [inline django_project/monitor/observation_views.py]: Please use datainput.get('latitude', 0)

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 4 min 57 sec

File: django_project/monitor/observation_views.py (hunk 1 above)
Comment (tinashechiraya) [reply django_project/monitor/observation_views.py]: @meomancer also if you check in network tab response ...it returns this when i trying adding the the_geom point field {
    "status": "error",
    "message": "Invalid parameters given for Point initialization."
} thats why I commented it out for now

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 6 min 1 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 56 sec

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 5 min 29 sec

**Review by dimasciput — APPROVED** [review]

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 6 sec

**Pull request merged**